tls/s2n_tls13_handshake.c

```diff
@@ -190,6 +190,8 @@ int s2n_tls13_handle_handshake_secrets(struct s2n_connection *conn)
 
     /* derive early secrets */
     GUARD(s2n_tls13_derive_early_secrets(&secrets, conn->psk_params.chosen_psk));
+    /* since early secrets have been computed, PSKs are no longer needed and can be cleaned up */
+    GUARD_AS_POSIX(s2n_psk_parameters_wipe(&conn->psk_params));
```

Comment:
No. `_wipe` will free the memory allocated for identities, secrets, etc., for individual PSKs. The only memory it doesn't free is the memory allocated for the PSK structures themselves. At the moment we only free that memory when we free the connection (`s2n_connection_free`); if we reuse the connection (`s2n_connection_wipe`), we reuse that memory. Calling `_free` here would override the `s2n_connection_wipe` behavior.

We could choose to changed the `s2n_connection_wipe` behavior to not reuse the PSK memory, but that would be a separate decision.

Reply:
What is the argument for allowing the reuse of PSK memory?

Reply:
The same argument for all the memory reuse in `s2n_connection_wipe` vs `s2n_connection_free`. Allocating new memory is expensive, so if it can be reused it should be reused.